lua/trader.lua

```diff
@@ -0,0 +1,406 @@
+	local confirm_query=confirm_queries[math.random(1,#confirm_queries)]
+	if(string.sub(item.image,7,10) == "book") then confirm_query = _"Hey, this ain't a library!  You wanna take that home to read?" end
+	local dialogDefinition = {
+		wml.tag.tooltip { id = "tooltip_large" },
+		wml.tag.helptip { id = "tooltip_large" },
+		wml.tag.grid {
+			wml.tag.row {
+				wml.tag.column {  -- column 1, item
+					border = "all",
+
+					border_size=50,
+					wml.tag.grid {
+						wml.tag.row { -- item image and name
+							wml.tag.column {
+								wml.tag.grid {
+									wml.tag.row {
+										wml.tag.column {
+											border = "bottom",
+											border_size=10,
+											wml.tag.image {
+												label = item.image
+											}
+										},
+										wml.tag.column {
+											wml.tag.label {
+												label = item.name .. "  (" .. item.sort .. ")"
+											}
+										},
+									}
+								}
+							}
+						},
+						wml.tag.row { -- item description
+							wml.tag.column {
+								wml.tag.label {
+									use_markup = true,
+									label = item.description
+								}
+							}
+						}
+					}
+				},
+				wml.tag.column {  -- column2, let's do business
+						wml.tag.grid {
+							wml.tag.row { -- the trader
+								wml.tag.column {
+									wml.tag.grid {
+										wml.tag.row {
+											wml.tag.column {
+												wml.tag.label {
+													border = "bottom",
+													border_size=30,
+													use_markup = true,
+													label = "<span size='large' color='wheat' weight='bold'>".. trader .."</span>"
+												}
+											},
+											wml.tag.column {
+												wml.tag.image {
+													label = "merchant.png"
+												}
+											}
+										}
+									}
+								}
+							},
+							wml.tag.row {  -- you want it?
+								wml.tag.column {
+										wml.tag.label {
+										use_markup = true,
+										label = "<span color='wheat' weight='bold'>".. confirm_query .."</span>"
+									}
+								}
+							},
+							wml.tag.row {  -- blank space
+								wml.tag.column {
+									wml.tag.spacer { height = 20 }
+								}
+							},
+							wml.tag.row {  -- the yes/no buttons
+								wml.tag.column {
+									wml.tag.grid {
+										wml.tag.row {
+											wml.tag.column {
+												border = "all",
+												border_size = 20,
+												wml.tag.button { id = "ok", label = _"Yes" },
+											},
+											wml.tag.column {
+												border = "all",
+												border_size = 20,
+												wml.tag.button {
+													id = "closeit",
+													label = _"No",
+													return_value_id = "cancel"
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+
+	local ret_val=false
+
+	local function preshow(dialog)
+		dialog.ok.on_button_click = function()
+			ret_val = true
+		end
+	end
+
+	local function postshow()
+	end
```

Comment:
This empty function isn't needed: `postshow` argument of `show_dialog()` is optional.